themes/qgrid/source/js/menu.js

```diff
@@ -15,26 +15,196 @@ function toggleVisibility(e) {
 	}
 }
 
+function replaceSymbol(text) {
+	return text.replace(/%20/g, ' ');
+}
+
+function searchOnLoad(e) {
+	e.stopImmediatePropagation();
+	const active = document.getElementsByClassName('active-topic')[0];
+	if (active) {
+		if (location.hash) {
+			localStorage.setItem('input', replaceSymbol(location.hash.substr(8)));
+		}
+		else {
+			if (localStorage.getItem('input')) {
+				location.hash = 'search=' + localStorage.getItem('input');
+			}
+		}
+		searchOnPage();
+	}
+	localStorage.removeItem('input');
+}
+
+window.onhashchange = function () {
+	searchOnPage();
+}
+
 function filter(e) {
 	e.stopImmediatePropagation();
 	const input = e.target || e.srcElement;
 	const filter = input.value.toLowerCase();
+	if (filter) { location.hash = 'search=' + filter; }
+	else { location.hash = ''; }
+	searchOnPage();
+	tagSearch();
+	selectSymbols();
+	localStorage.setItem('input', location.hash.substr(8));
+}
+
+function searchOnPage() {
+	const h2 = document.getElementsByTagName('h2');
+	const p = document.getElementsByTagName('p');
+	const code = document.getElementsByTagName('code');
+	const text = [...h2, ...p, ...code];
+	for (let i = text.length - 1; i >= 0; i--) {
+		const title = text[i].textContent;
+		const symbol = title.split('');
```

Comment:
what does mean symbol, please try to think about names that are readable for the next developers :)